### PR TITLE
chore: Refactor deployment to dev/prod environments

### PR DESCRIPTION
**Pull request by atomrc**
The code for deploying to dev or prod env is a copy/paste for every single env. 
This PR refactors this and deduce the env to deploy to depending on the branch or the tag and avoids code duplication

### CODE REVIEW

File: .github/workflows/test_build_deploy.yml [hunk 1]
```diff
@@ -309,6 +223,7 @@ jobs:
         with:
           tag_name: ${{env.TAG}}
           name: ${{env.TAG}}
+          body_path: ./CHANGELOG.md
```
Comment (atomrc) [inline .github/workflows/test_build_deploy.yml:228]: This will add the changelog automatically to the github release

Comment (tlebon) [reply .github/workflows/test_build_deploy.yml:228]: is this not happening on web? it happens on tm but maybe its handled elsewhere I'm not sure now.

Comment (tlebon) [reply .github/workflows/test_build_deploy.yml:228]: yes this is included on TM, just checked.

**Review by tlebon — APPROVED** [review]
Seems like a nice change, ill have to incorporate this into TM as well.

File: .github/workflows/test_build_deploy.yml (hunk 1 above)
Comment (atomrc) [reply .github/workflows/test_build_deploy.yml:228]: It was happening for production releases, but not staging for some reasons ... 🙃

**Pull request merged**